### PR TITLE
logger: Add spaces after comma when printing containers

### PR DESCRIPTION
**Pull request by Hespian**
This makes the output easier to read for humans

### CODE REVIEW

**Comment by lorenzhs** [comment]
Meh, I don’t want this in many cases. What if the spacing logger had spaces between items and the non-spacing one remained as it is? That would be a more complicated fix but I think it would be preferable.

**Comment by Hespian** [comment]
I like that idea. The new commit should do that.

File: tlx/logger/core.hpp [hunk 1]
```diff
@@ -231,7 +231,8 @@ template <typename AnyType>
 class LoggerFormatter<AnyType>
 {
 public:
-    static void print(std::ostream& os, const AnyType& t) {
+    static void print(std::ostream& os, const AnyType& t, const bool addSpace) {
+	(void)addSpace;
```
Comment (lorenzhs) [inline tlx/logger/core.hpp]: why not just `const bool /* addSpace */` and save the cast to void?

File: tlx/logger/map.hpp [hunk 2]
```diff
@@ -39,15 +40,16 @@ template <typename K, typename V, typename C, typename A>
...
-    static void print(std::ostream& os, const std::multimap<K, V, C, A>& data) {
+    static void print(std::ostream& os, const std::multimap<K, V, C, A>& data, const bool addSpace) {
         os << '{';
         for (typename std::multimap<K, V, C, A>::const_iterator it = data.begin();
              it != data.end(); ++it)
         {
-            if (it != data.begin()) os << ", ";
-            LoggerFormatter<K>::print(os, it->first);
+            if (it != data.begin()) os << ',';
+            if (it != data.begin() && addSpace) os << ' ';
```
Comment (lorenzhs) [inline tlx/logger/map.hpp]: `if (it != data.begin()) os << (addSpace ? ", " : ",");`?

File: tlx/logger/core.hpp (hunk 1 above)
Comment (bingmann) [reply tlx/logger/core.hpp]: err, this won't fly. extra bool for just one type? nope.

**Comment by bingmann** [comment]
Okay, so there are two ways to actually get this merged:
I think one could make a separate include "logger/pair_comma.hpp" which adds a comma only for std::pair. I don't see why one should waste space for vector etc.
Or make a global define option TLX_LOGGER_COMMA_SPACE or such.

**Comment by lorenzhs** [comment]
Hostile fork in 3, 2, 1, ...

File: tlx/logger/core.hpp (hunk 1 above)
Comment (lorenzhs) [reply tlx/logger/core.hpp]: it's for every container type, not just one.

**Comment by lorenzhs** [comment]
I would like to suggest a third option for when to add space and when not to. The idea of aidding spaces is to make it easier to visually find the boundaries between the items. Assume you have a vector of tuples. Then wouldn't you want spaces only between the items of the vector but not the items of the spaces, i.e., don't propagate `addSpace` down?

**Comment by bingmann** [comment]
Yes, also possible.

**Comment by bingmann** [comment]
a) don't overbloat the logger code.
b) don't change old behaviour.

**Pull request closed without merging**